File: 240322/부분 수열의 합이 k/the-sum-of-the-subsequences-is-k.java
import java.util.Scanner;

public class Main {
    private static final Scanner sc = new Scanner(System.in);

    public static void main(String[] args) {
        int n = sc.nextInt();
        int k = sc.nextInt();
        int[] arr = new int[n + 1];
        for (int i = 1; i <= n; i++) {
            arr[i] = sc.nextInt();
        }

        int[] sum = new int[n + 1];
        for (int i = 1; i <= n; i++) {
            sum[i] = sum[i - 1] + arr[i];
        }

        int cnt = 0;
        for (int i = n; i >= 1; i--) {
            for (int j = 1; j <= i; j++) {
                int result = sum[i] - sum[j - 1];

                if (result == k) {
                    cnt++;
                }
            }
        }
        System.out.println(cnt);
    }
}

File: 240409/부분 수열의 합이 k/the-sum-of-the-subsequences-is-k.kt
import java.util.*

val sc = Scanner(System.`in`)

val n = sc.nextInt()
val k = sc.nextInt()
val arr = IntArray(n + 1)
val sums = IntArray(n + 1)

fun main() {
    for (i in 1..n) {
        arr[i] = sc.nextInt()
    }

    for (i in 1..n) {
        sums[i] = sums[i - 1] + arr[i]
    }

    var answer = 0
    for (e in 1..n) {
        for (s in 1..e) {
            val sum = sums[e] - sums[s - 1]
            
            when (sum) {
                k -> answer++
                else -> { }
            }
        }
    }

    print(answer)
}